nomad/variables_endpoint.go

```diff
@@ -344,15 +365,17 @@ func (sv *Variables) Read(args *structs.VariablesReadRequest, reply *structs.Var
 			// Setup the output
 			reply.Data = nil
 			if out != nil {
+
+				if !(aclObj != nil && aclObj.IsManagement()) {
+					out.Lock = nil
+				}
```

Comment:
This is modifying the original value we got out of the state store outside of a Raft transaction (and in a read request!). We can only modify copies of objects we get out of the state store, so I don't think we can make this change.